src/HttpHandlers/LfdsUsernamePasswordHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Mime;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Laserfiche.Api.Client.Lfds;
+
+namespace Laserfiche.Api.Client.HttpHandlers
+{
+    /// <summary>
+    /// LFDS username password handler to set the authorization header given username, password, organization, repository ID and base URL.
+    /// </summary>
+    public class LfdsUsernamePasswordHandler : IHttpRequestHandler
+    {
+        private string _accessToken;
+        private readonly string _username;
+        private readonly string _password;
+        private readonly string _baseUrl;
+        private readonly string _repoID;
+        private readonly string _organization;
+
+        public LfdsUsernamePasswordHandler(string username, string password, string organization, string repoID, string baseUrl)
+        {
+            _username = username;
+            _password = password;
+            _baseUrl = baseUrl.TrimEnd('/');
+            _repoID = repoID;
+            _organization = organization;
+        }
+
+        public async Task<BeforeSendResult> BeforeSendAsync(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(_accessToken))
+            {
+                var client = new AccessTokensApiClient(new HttpClient { BaseAddress = new Uri(_baseUrl)});

```

Comment:
probably better to not create a new AccessTokensApiClient every time we get a new access token. Can probably re use it